NuRadioReco/modules/io/rno_g/readRNOGDataMattak.py

```diff
@@ -0,0 +1,569 @@
+   """
+   Simple baseline correction function. Determines baseline in discrete chuncks of "n_bins" with
+   the function specified (i.e., mean or median).
+   
+   Parameters
+   ----------
+   
+   wfs: np.array(n_events, n_channels, n_samples)
+      Waveforms of several events/channels.
+      
+   n_bins: int
+      Number of samples/bins in one "chunck". If None, calculate median/mean over entire trace. (Default: 128)
+      
+   func: np.mean or np.median
+      Function to calculate pedestal
+   
+   Returns
+   -------
+   
+   wfs_corrected: np.array(n_events, n_channels, n_samples)
+      Baseline/pedestal corrected waveforms
+   """
+    
+   # Example: Get baselines in chunks of 128 bins
+   # wfs in (n_events, n_channels, 2048)
+   # np.split -> (16, n_events, n_channels, 128) each waveform split in 16 chuncks
+   # func -> (16, n_events, n_channels) pedestal for each chunck
+   if n_bins is not None:
+      baseline_values = func(np.split(wfs, 2048 // n_bins, axis=-1), axis=-1)
+
+      # np.repeat -> (2048, n_events, n_channels) concatenate the 16 chuncks to one baseline
+      baseline_traces = np.repeat(baseline_values, n_bins % 2048, axis=0)
+   else:
+      baseline_values = func(wfs, axis=-1)
+      # np.repeat -> (2048, n_events, n_channels) concatenate the 16 chuncks to one baseline
+      baseline_traces = np.repeat(baseline_values, 2048, axis=0)
+          
+   # np.moveaxis -> (n_events, n_channels, 2048)
+   baseline_traces = np.moveaxis(baseline_traces, 0, -1)
+    
+   return wfs - baseline_traces
+
+
+def get_time_offset(trigger_type):
+   """ 
+   Mapping the offset between trace start time and trigger time (~ signal time). 
+   Temporary use hard-coded values for each trigger type. In the future this
+   information might be time, station, and channel dependent and should come 
+   from a database (or is already calibrated in mattak)
+   
+   Current values motivated by figures posted in PR https://github.com/nu-radio/NuRadioMC/pull/519
+   
+   Parameters
+   ----------
+   
+   trigger_type: str
+      Trigger type encoded as string from Mattak
+      
+   Returns
+   -------
+   
+   time_offset: float
+      trace_start_time = trigger_time - time_offset 
+   
+   """
+   
+   time_offsets = {
+      "FORCE": 0,
+      "LT": 250 * units.ns,  
+      "RADIANT": 475 * units.ns  
+   }
+   
+   if trigger_type.startswith("RADIANT"):
+      trigger_type = "RADIANT"
+   
+   if trigger_type in time_offsets:
+      return time_offsets[trigger_type]
+   else:
+      raise KeyError(f"Unknown trigger type: {trigger_type}. Known are: FORCE, LT, RADIANT. Abort ....")
+
+
+class readRNOGData:
+
+   def begin(self, 
+             data_dirs, log_level=logging.INFO, selectors=None, 
+             read_calibrated_data=False,
+             apply_baseline_correction=True,
+             convert_to_voltage=True,
+             select_triggers=None,
+             select_runs=True,
```

Comment:
While it's nice to have this  many options, I think this is also a bit overwhelming for someone that 'just' wants to read, say, a single combined.root file that I got from a more senior colleague. Some suggestions:
- Rearrange the parameters to have the more relevant/frequently used ones first. E.g. `log_level` shouldn't be the second parameter...
- (docstring) - move some of the more obscure parameters to an `Other Parameters` section
- (low priority) - maybe passing (a list of) combined.root files should also work? Though this may just be my bias because 'that's how we used to do it', I can see the advantages of the way you've implemented it.

Reply:
To your last point: I think this would require changes in Mattak. I opened an issue https://github.com/RNO-G/mattak/issues/7 to discuss this. FFTJ

Reply:
I rearranged it a bit. Please have a look

Reply:
((in fact: the last point is more or less in-line with a previous comment I had... I would have been more radical and not made run-selection at all a part of the reader, but since you can just pass the directory without the other optional selection this should work great as is.))

Passing explicitly a filenames=["combined.root"] file like before I find useful ... not sure if the way the mattak class is called already would work now if you pass it a filename instead of a dir? Otherwise should be quick to include buy might needs some if/else checking depending on what is passed.

Reply:
yes mattak can not handle that atm.